R/utils.R

```diff
@@ -9,6 +9,21 @@ check_no_value_is_missing <- function(data, column) {
   invisible(data)
 }
 
+filter_and_warn_na <- function(data, column) {
+  if (anyNA(data[[column]])) {
+    name_dataset <- deparse(substitute(data))
+    warning_message = paste("Removing rows in", name_dataset, "where `{column}` is NA")
+    warn(
+      glue(warning_message),
+      class = "na_crucial_economic_input"
+    )
+
+    data <- filter(data, !is.na(.data[[column]]))
+
+    return(data)
+  }
```

Comment:
So the issue is that you need to put this `return(data)` outside of the `if()` statement. 

What is happening now is that, if the input `data` passes `anyNA`, the function works as expected, but if it doesn't, the function returns NULL, and that's why all the tests are failing.

Reply:
Makes sense, thanks for checking!